### PR TITLE
[MERGE][FIX] livechat, *: open livechat from URL

### PR DESCRIPTION
**Pull request by msh-odoo**
*: mail, web

This fixes the following issues:

### 1. Livechat menu was not visible when accessing from the customer URL

The customer URL is obtainable by following those steps:

- Access `/web`
- Click on Live Chat app
- Open a Live Chat Channel
- Click on "Widget" sheet
- Access livechat from (2nd) "Copy Text" button

This is due to missing JS modules in the external lib, such as the root widget.

### 2. The position of the buttons "Copy Text" is modified on pressing 'ALT'

This is caused by the keyboard navigation overlay that changes the CSS
position of the parent of the overlay element. It sets the position to
`relative`, so that the overlay exactly covers the element. However, if the
element had a position `absolute`, changing it to `relative` may re-position it
somewhere else on the page.

In fact, it is not necessary to set the CSS position to `relative` in this case,
because the overlay has CSS position `absolute`: an element with `absolute`
will match the size of the parent if the parent is also `absolute`.

Task-ID 1885445

### CODE REVIEW

**Comment by alexkuhn** [comment]
@VincentSchippefilt Could you please just check the commit about keyboard navigation? Thanks!

File: addons/web/static/tests/chrome/keyboard_navigation_mixin_tests.js [hunk 1]
```diff
@@ -27,5 +27,37 @@ QUnit.module('KeyboardNavigationMixin', function () {
...
+        $button.css('position', 'absolute');
+
+        // minimum set of attribute to generate a native event that works with the mixin
+        var e = new Event("keydown");
+        e.key = '';
+        e.altKey = true;
+        w.$el[0].dispatchEvent(e);
+
+        assert.strictEqual($button.css('position'), 'absolute',
+            "should not have kept the CSS position of the button");
```
Comment (aab-odoo) [inline addons/web/static/tests/chrome/keyboard_navigation_mixin_tests.js]: w.destroy() !!!!

File: addons/web/static/src/js/chrome/public_root_widget.js [hunk 2]
```diff
@@ -13,10 +14,30 @@ var Widget = require('web.Widget');
...
+     *
+     * @private
+     * @param {OdooEvent} ev
+     * @param {function} [ev.data.callback]
+     */
+    _onGetSession: function (ev) {
+        if (ev.data.callback) {
+            ev.data.callback(session);
+        }
+    },
```
Comment (aab-odoo) [inline addons/web/static/src/js/chrome/public_root_widget.js]: This whole session thing accessible by trigger was an error of design. Are there many uses of session in the external livechat? If not, you could directly require web.session instead

File: addons/web/static/tests/chrome/keyboard_navigation_mixin_tests.js (hunk 1 above)
Comment (alexkuhn) [reply addons/web/static/tests/chrome/keyboard_navigation_mixin_tests.js]: Former test in this file was not destroying this widget.
I have added mocked environment + clean destroy.

**Comment by aab-odoo** [comment]
robodoo r+

File: addons/web/static/tests/chrome/keyboard_navigation_mixin_tests.js (hunk 1 above)
Comment (VincentSchippefilt) [reply addons/web/static/tests/chrome/keyboard_navigation_mixin_tests.js]: @alexkuhn you are awesome !

**Pull request closed without merging**